Domains/2-DisciplinePhysical/Piping/PipeworkPhysical.ecschema.xml

```diff
@@ -0,0 +1,244 @@
+        <Source multiplicity="(0..*)" roleLabel="groups" polymorphic="true">
+            <Class class="PipingSystem"/>
+        </Source>
+        <Target multiplicity="(0..*)" roleLabel="is grouped by" polymorphic="true">
+            <Class class="IPipingElement"/>
+        </Target>
+    </ECRelationshipClass>
+
+    <ECEntityClass typeName="IPipingPhysicalType" modifier="Abstract" displayLabel="Piping Physical Type" description="An interface that can be mixed-into a bis:PhysicalType that defines a shared set of properties whose values vary per-type of IPipingElement rather than per-instance.">
+        <BaseClass>dsys:IDistributionElementType</BaseClass>
+        <ECCustomAttributes>
+            <IsMixin xmlns="CoreCustomAttributes.01.00.03">
+                <AppliesToEntityClass>bis:PhysicalType</AppliesToEntityClass>
+            </IsMixin>
+        </ECCustomAttributes>        
+    </ECEntityClass>
+
+    <ECEntityClass typeName="PipeElement" modifier="Abstract" displayLabel="Pipe Element" description="A bis:PhysicalElement used to typically join two sections of a piping network.">
+        <BaseClass>bis:PhysicalElement</BaseClass>
+        <BaseClass>IPipingElement</BaseClass>
+        <BaseClass>dsys:IDistributionFlowElement</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="PipeElementType" modifier="Abstract" displayLabel="Pipe Element Type" description="Defines a shared set of properties whose values vary per-type of PipeElement rather than per-instance.">
+        <BaseClass>bis:PhysicalType</BaseClass>
+        <BaseClass>IPipingPhysicalType</BaseClass>
+        <ECProperty propertyName="GrossWeight" typeName="double" kindOfQuantity="AECU:WEIGHT" displayLabel="Gross Weight" description="Total Gross Weight of the object without any add-on parts and not taking into account possible processing features (cut-out's, etc.) or openings and recesses. Not including contained matter."/>
+        <ECProperty propertyName="PressureRange" typeName="bis:BoundedDoubleValue" kindOfQuantity="AECU:PRESSURE" displayLabel="Pressure Range" description="Allowable maximum, minimum and typical working pressure (relative to ambient pressure)."/>
+        <ECProperty propertyName="TemperatureRange" typeName="bis:BoundedDoubleValue" kindOfQuantity="AECU:TEMPERATURE" displayLabel="Temperature Range" description="Allowable maximum, minimum and typical temperature."/>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="Pipe" modifier="None" displayLabel="Pipe" description="A straight PipeElement used to typically join two sections of a piping network.">
+        <BaseClass>PipeElement</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="PipeType" modifier="None" displayLabel="Pipe Type" description="Defines a shared set of properties whose values vary per-type of Pipe rather than per-instance.">
+        <BaseClass>PipeElementType</BaseClass>
+    </ECEntityClass>
```

Comment:
this is inconsistent with PipeFittingType when it comes to IFC. i.e. pipefittingType and subclasses map quite well to:
https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/link/ifcpipefittingtypeenum.htm 

however, PipeElementType doesn't map to:
https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/link/ifcpipesegmenttypeenum.htm

at all. i would also rename PipeElementType to PipeType (otherwise it sounds like a PipeFitting would also be a PipeElementType, which it's not

Reply:
The setup for the intermediate PipeElement/PipeElementType classes was introduced in order to accommodate Flexible pipes. The latter were not part of the requirements for mapping into StructuralAnalysis, but some external apps support them (e.g. Revit & OBD). Since they are uncommon, we're leaving them out from v1.0.0 but we envision their introduction in a subsequent release.